django/db/models/expressions.py

```diff
@@ -532,13 +544,10 @@ def identity(self):
         next(arguments)
         identity = [self.__class__]
         for arg, value in arguments:
```

Comment:
The thing we missed when implemented this logic is that `Expression.__init__` overrides like `Concat.__init__(self, *expressions, **extra)` for a call of the form `Concat("foo", "bar", output_field=TextField())` will have the following `arguments`

```python
[("expressions", ("foo", "bar")), ("extra", {"output_field": TextField())]
```

but we need `TextField()` to be turned into `TextField` (its type) so proper equality checks to prevail so `*args` and `**kwargs` captures must be _expanded_.